experimenter/experimenter/experiments/models.py

```diff
@@ -1306,30 +1308,108 @@ def timeline(self):
 
         return timeline_entries
 
-    @property
-    def metric_areas(self):
-        metric_areas = NimbusConstants.DEFAULT_METRIC_AREAS.copy()
+    def get_metric_areas(
+        self, analysis_basis, segment, reference_branch, window="overall"
+    ):
+        metric_areas = {}
+
+        metric_areas[NimbusUIConstants.NOTABLE_METRIC_AREA] = []
+        metric_areas[NimbusUIConstants.KPI_AREA] = self.get_kpi_metrics(
+            analysis_basis, segment, reference_branch, window
+        )
+
+        metrics_metadata = (
+            self.results_data.get("v3", {}).get("metadata", {}).get("metrics")
+            if self.results_data
+            else {}
+        ) or {}
+        all_outcome_metric_slugs = []
+        for slug in self.primary_outcomes + self.secondary_outcomes:
```

Comment:
```suggestion
        for slug in itertools.chain(self.primary_outcomes, self.secondary_outcomes):
```

if we use `itertools.chain()` we don't actually construct a new list.